tests/CMakeLists.txt

```diff
@@ -0,0 +1,12 @@
+add_subdirectory(catch)
```

Comment:
ah I was hoping to keep the very-outdated `catch` folder _inside_ `tests/deprecated`, and use CMake to obtain the current version of Catch2 for the v4 tests, as discussed [here](https://github.com/QuEST-Kit/QuEST/issues/514). 

I can wrangle with this myself though if it proves a fuss - in that case, let's keep `catch` inside `tests/deprecated/` and replace the above with `add_subdirectory(deprecated/catch)` so it still builds for now

Reply:
Ah so I would rather not run two different versions of Catch2 at the same time, which is what we would be doing here with the deprecated API tests running Catch2 v2.7 and the new tests running Catch2 v3.x, and potentially both sitting in the same folder if the user doesn't already have Catch2 v3.x installed somewhere findable. 

Looking at your comment on #514 I see that you wanted to use some of the new v3.x functionality, so perhaps it makes sense to just bump everything to 3.x now and do away with the headers? From experience I think it should be easy enough, but I'll get through the rest of these comments before I try it!

Reply:
Working on it on a separate branch [here](https://github.com/otbrown/QuEST/tree/catch2-v3). Mostly plain sailing except some _weirdness_ with the Generators, which I'll look into more tomorrow. Have most of a fix already though, so won't be a blocker!

Reply:
True, definitely gross to mix versions of Catch yea - I was envisioning migrating the old tests to new Catch was going to be an undue fuss, but brilliant if it works! No surprise the generator design has changed with Catch2 v3 hence my [lazy stubs](https://github.com/QuEST-Kit/QuEST/blob/334a128ad9420fcf1cbfb9387782f13d9bf4f630/tests/utils/lists.cpp#L38) hehe.

Reply:
Done!